crates/type-system/src/ontology/data_type/mod.rs

```diff
@@ -94,11 +74,38 @@ impl DataType {
     }
 }
 
-#[cfg_attr(target_arch = "wasm32", derive(Tsify))]
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
-#[serde(deny_unknown_fields)]
+impl FromStr for DataType {
+    type Err = ParseDataTypeError;
+
+    fn from_str(data_type_str: &str) -> Result<Self, Self::Err> {
+        let data_type_repr: repr::DataType = serde_json::from_str(data_type_str)
+            .map_err(|err| ParseDataTypeError::InvalidJson(err.to_string()))?;
```

Comment:
Do we want to classify the JSON error further? This might be useful for the user as it's a difference, if the JSON is malformed or if e.g. a list was expected but an object was passed.

Reply:
**At the moment** I don't think it's worth the time investment. JSON errors shouldn't really be possible if people are using the TypeScript types (or Rust types, or any other bindings we may produce)